File: CareConnect/CareConnect.Models/Responses/Employee.cs
using System;
using System.Collections.Generic;

namespace CareConnect.Models.Responses
{
    public class Employee
    {
        public DateTime HireDate { get; set; }

        public DateTime? EndDate { get; set; }

        public bool Employed => EndDate == null; 

        public string JobTitle { get; set; } = null!;

        public DateTime ModifiedDate { get; set; }

        public virtual User User { get; set; } = null!;

        public virtual Qualification? Qualification { get; set; }
    }
}

File: UI/careconnect_admin/lib/models/responses/employee.dart
import 'package:careconnect_admin/models/responses/qualification.dart';
import 'package:careconnect_admin/models/responses/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'employee.g.dart';

@JsonSerializable(explicitToJson: true)
class Employee {
  final DateTime hireDate;
  final DateTime? endDate;
  final String jobTitle;
  final DateTime modifiedDate;
  final User? user;
  final Qualification? qualification;
  // List<EmployeeAvailability> employeeAvailabilities;

  bool get employed => endDate == null;

  Employee({
    required this.hireDate,
    this.endDate,
    required this.jobTitle,
    required this.modifiedDate,
    this.user,
    this.qualification,
    // required this.employeeAvailabilities,
  });

  factory Employee.fromJson(Map<String, dynamic> json) =>
      _$EmployeeFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeToJson(this);
}
